### ISSUE TITLE
Some questions about license changing.

### ISSUE THREAD

**Issue by JayLiangs** [icsharpcode/SharpZipLib#803]
Hello,
SharpZipLib Team

I'm a law school student，and I’m doing legal research about opensource license and copyright system.
I know that from your Readme and Wiki that you have  changed the license of sharpziplib from GPL to LGPL, then to MIT.
My questions are:
1. why you chose the GPL with classpath exception as the original license. Is it just because ported from Classpath Project?
2. Since you changing the license to permissive MIT license, can I deem that the team never wish force the user opensource their project（which may linked with sharpziplib） , even before 2016？

Thanks for your reply.

**Comment by christophwille** [comment]
Yes, the original license was chosen because at that time it was a simple and very direct Java port (we wanted to use it for SharpDevelop to efficiently store documentation XMLs for code completion)

The change to MIT was part (or a a follow on) to the license change in SharpDevelop. Having a "non-standard" license (GPL with exeception) can be confusing, and we wanted to make it clear that linking is a-ok. More generally, GPL is fine for standalone tools, whereas when it comes to libraries, the license shouldn't create an adoption barrier. And quite frankly adoption (use) of the library is the most important thing for a thriving OSS project.

**Comment by JayLiangs** [comment]
Thanks a lot. 

I still have a further question to bother you.
Q: If I make a .dll to incorporate SharpZipLib with my program, more specifically, I compile your Sharpziplib with my code into a .dll and then use the main program to call the dll. I think because the way of use is dll(dynamic link library), it is consistent with the classpath exception. And there is no need to opensource my "main program". Is my interpretation right?

Thanks for your reply.

**Comment by christophwille** [comment]
With today's MIT codebase you can basically use it however you want (NuGet or code)

**Comment by piksel** [comment]
@JayLiangs regarding the original license, we didn't write it and are even less lawyers than you, so I don't think we can give you a better answer to that.
